File: src/main/java/org/example/coin_laundry_app_backend/geo/application/service/ReverseGeocodingService.java
package org.example.coin_laundry_app_backend.geo.application.service;

import org.example.coin_laundry_app_backend.geo.application.service.model.ReverseGeoModel;
import org.example.coin_laundry_app_backend.geo.domain.model.EPSG4326Coordinate;
import reactor.core.publisher.Mono;

public interface ReverseGeocodingService {

    Mono<ReverseGeoModel> getReverseGeocoding(EPSG4326Coordinate coordinate);
}

File: carry-geo/src/main/kotlin/com/carry/geo/application/service/ReverseGeocodingService.kt
package com.carry.geo.application.service

import com.carry.geo.application.port.inbound.ReverseGeocodingUseCase
import com.carry.geo.application.port.outbound.ReverseGeocodingPort
import com.carry.geo.domain.model.ReverseGeocodingResult
import com.carry.geo.domain.vo.Coordinate
import org.springframework.stereotype.Service

@Service
class ReverseGeocodingService(
    private val reverseGeocodingPort: ReverseGeocodingPort,
) : ReverseGeocodingUseCase {

    override fun reverseGeocode(coordinate: Coordinate): ReverseGeocodingResult {
        return reverseGeocodingPort.reverseGeocode(coordinate)
    }
}
